### ISSUE TITLE
add coco-text as a test/train set

### ISSUE THREAD

**Issue by Thomas-MMJ** [mindee/doctr#1131]
### 🚀 The feature

You might consider adding COCO-text as one of the supported datasets,

https://vision.cornell.edu/se3/coco-text-2/#download

### Motivation, pitch

It is another high quality dataset, text on objects at various angles (sides of vehicles, signs, etc.)

**Comment by felixdittrich92** [comment]
Hey @Thomas-MMJ 👋 ,

Thanks for the request do you want to add it maybe ?
If so im happy to guide you If there is any help needed :)

**Comment by dvando** [comment]
Hi @felixdittrich92 , has anybody worked on this? I'd love to hop into the project and contribute to this issue. :)

**Comment by felixdittrich92** [comment]
Hey @dvando 👋,

No it's still open.
Sure feel free to work on it, if you have any questions or need some help contact me :)

**Comment by dvando** [comment]
Hi @felixdittrich92 , my apology it took me a while to actually work on it, I've been dealing with some issues from work. 

I've got some questions about the URLs for download, COCO-text has 2 separate URLs, the first one is for the [images](http://images.cocodataset.org/zips/train2014.zip), and the second is for the [labels](https://github.com/bgshih/cocotext/releases/download/dl/cocotext.v2.zip), but the `VisionDataset` only accepts 1 URL which I believe lead to a compressed images and it's labels. 

I also checked the other datasets (funsd, cord, synttext, etc), and all of them initialized the `VisionDataset` using 1 URL only, I was thinking about merging the files myself, but then I was wondering if that's the right thing to do. (Changing the base class should not be an option I believe)

Sorry, and thanks in advance. :)

**Comment by felixdittrich92** [comment]
Hi @dvando :smile: 
No stress ^^ 

Option 1: You could take a look at https://github.com/mindee/doctr/blob/main/doctr/datasets/imgur5k.py (here the user needs to provide the paths to the data and we provide only the loader)
Option 2: What's the dataset size in MB / GB ? What's the license ? If both isn't troublesome we could combine the dataset and upload it :)

**Comment by dvando** [comment]
So with option 1, the user should download the images and the labels by themself? That sounds okay.
The dataset has ~13 GB in size and has [CC by 4.0](https://creativecommons.org/licenses/by/4.0/legalcode) license.

Both sound fine to me, which one do you prefer @felixdittrich92 ? :)

**Comment by felixdittrich92** [comment]
> So with option 1, the user should download the images and the labels by themself? That sounds okay. The dataset has ~13 GB in size and has [CC by 4.0](https://creativecommons.org/licenses/by/4.0/legalcode) license.
> 
> Both sound fine to me, which one do you prefer @felixdittrich92 ? :)

Option 1 :+1:

**Comment by felixdittrich92** [comment]
As reference PR: https://github.com/mindee/doctr/pull/1359  :)

**Comment by sarjil77** [comment]
@felixdittrich92, 
i would like to work on this and want to contribute over here, please guide me over here

thanks in advance.

**Comment by felixdittrich92** [comment]
Sure @sarjil77 :)

First download the dataset: https://bgshih.github.io/cocotext/ (annotations) & images http://images.cocodataset.org/zips/train2014.zip

The following PR can be used as reference: https://github.com/mindee/doctr/pull/1359/files

In `doctr/datasets` create a new python file `coco_text.py`, which contains the conversion logic ref.: https://github.com/mindee/doctr/blob/main/doctr/datasets/wildreceipt.py
For `detection_task=True` it returns only the det converted det annotations as the name says use_polygons it returns 4point polygon coordinates otherwise boxes and with `recognition_task=True` it crops all the polygons/boxes on the fly to create a recognition dataset otherwise if `detection_task=False` and `recognition_task=False` it returns the OCR E2E dataset containing the polygon/box annotations and corresponding labels.

In `tests/conftest.py` we need to create a mock of the original annotations for testing purposes for example:
https://github.com/mindee/doctr/blob/ebfc9f3b6f85ceff96502f834c4f8b01327d6fcd/tests/conftest.py#L675

These fixture is used in https://github.com/mindee/doctr/blob/ebfc9f3b6f85ceff96502f834c4f8b01327d6fcd/tests/pytorch/test_datasets_pt.py#L741
and https://github.com/mindee/doctr/blob/ebfc9f3b6f85ceff96502f834c4f8b01327d6fcd/tests/tensorflow/test_datasets_tf.py#L714
You can place the `coco_text` tests directly after the wildreceipt test cases :)

As last step we add the documentation entries:
See the first two modified files here: https://github.com/mindee/doctr/pull/1359/files

If you need any further information feel free to ask 👍

**Comment by sarjil77** [comment]
Thanks @felixdittrich92, will further look into it.

**Comment by sarjil77** [comment]
hey @felixdittrich92 , sorry i was busy with tedious work. 

i have added the script and when i am seeing the dataset i do not see any text or annotated text in json annotation file. Could you please verify that.

**Comment by sarjil77** [comment]
hey @felixdittrich92, sorry i was busy with tideous work.

i want to know, if i am seeing in wildreceipt their annotation and their labels format is different, so for coco_test it is also different. 

for wildreceipt:  {"box": [274.0, 308.0, 326.0, 308.0, 326.0, 286.0, 274.0, 286.0], "text": "$4.48", "label": 19},
for coco : we only have box coordinate i.e: 4 for eg: {"box": [347.9,93.6,196.7,35.6], "text": "PROGRESSIVE"}

so your guidance would be helpful.

**Comment by felixdittrich92** [comment]
Hi @sarjil77 👋,

Correct the mentioned wildrecipt PR is only meant as reference which parts needs to be implemented and how (e.g.: `use_polygons=True | False`, `detection_task=True | False`, `recognition_task=True | False`)

with `use_polygons=False` we need to append the following format : `xmin, ymin, xmax, ymax`
with `use_polygons=True`: `[[x1, y1], [x2, y2], [x3, y3], [x4, y4]]` 

So what you need to do is to implement the annotation loading and the coordinate conversion depending on the given parameters

So first you should understand which format is used in the annotations: [x, y, width, height] / [xmin, ymin, xmax, ymax] / .. ?

Does this help ? :)

Otherwise try to implement as much as possible and open a PR where we can iterate 👍
